### ISSUE TITLE
Black Morass – Antnormi does not respect rogue tank avoidance, Mossheart and elemental foes are immune to Agitatin Poison

### ISSUE THREAD

**Issue by RUTHINAS** [slowtorta/turtlewow-bug-tracker#5807]
**Please, provide a clear description what the bug is:**
I was tanking Black Morass on my rogue earlier, and Antnormi landed (at least) 9 consecutive hits through my avoidance, despite being unhittable on paper. This happened while I was not stunned, nor facing away from the boss, and I kept attacking the boss during this time while the healer did everything in their power to keep me alive (which was doomed to fail).

I am not certain of the exact course of the fight, as my combat log wasn't scrollable far enough, but it _might_ have something to do with the fact that I used an engineering Masterwork Training Dummy to avoid the stun/fear effect the boss does, by activating it mere moments before the cast went through.

If this has something to do with the somewhat recent change to rogue avoidance being nullified while soloing bosses (or doing them in small groups), in our case we had a full group of 5 people. Besides, shouldn't that change be retired already, since after Flourish was given a cooldown it should no longer be possible to maintain full unhittability for an extended period of time.

Additionally, the elemental mobs of the dungeon and the boss Mossheart are all immune to Agitating Poison, making tanking them much more difficult for rogues. I get the lore-based reasoning of having certain mobs be immune to certain damage types, but in this case they're being made immune to a specific tank (though admittedly very niche). Perhaps Agitating Poison could be made to generate its threat regardless of immunity to the damage component?

**Steps to reproduce the behavior with as much detail as possible:**

1. Tank Antnormi as a rogue
2. (Possibly: use Target Dummy)
3. Use whatever avoidance cooldowns you have
4. Observe the boss mauling through them like your avoidance does not exist

**IDs of affected creatures, items, quests or spells with a link to the relevant page:**

NPC: https://database.turtle-wow.org/?npc=65125

**Expected behavior. Describe how it should work:**
Bosses should respect avoidance, especially during ordinary runs.
Bosses should not be made immune to the threat mechanics of specific tanks, unless in a raid environment where it is reasonable to assume someone else can commit themselves to tanking an encounter. 5-person dungeon groups don't necessarily have anyone that could stand in for a tank that can not tank a fight due to design choices.
_Possibly:_ The rogue avoidance nullification effect should be retired, as it is no longer needed after further changes have been introduced to rogue tanking.

**Comment by twowTurtReynolds** [comment]
Works fine on nonrogue
![image](https://github.com/slowtorta/turtlewow-bug-tracker/assets/151180101/09eec794-91a2-469f-ace0-63ef2240ad10)

But not rogue
![image](https://github.com/slowtorta/turtlewow-bug-tracker/assets/151180101/19167d2a-162c-487f-a588-6d5c953626f0)
![image](https://github.com/slowtorta/turtlewow-bug-tracker/assets/151180101/f422ba2b-e75c-4dfc-97fe-f985bfe4444c)

Seems all mobs marked Boss do this:
![image](https://github.com/slowtorta/turtlewow-bug-tracker/assets/151180101/bbc1768e-6535-406f-bfca-e386419cd9b9)

The intention from the developers before was to stop solo rogue tanking on just Raid Bosses, but nothing about Dungeon mobs marked boss.

**Comment by twowTurtReynolds** [comment]
As for the agitating poison, that is not a bug but a suggestion, post that on the forums instead.

**Comment by RUTHINAS** [comment]
>The intention from the developers before was to stop solo rogue tanking on just Raid Bosses, but nothing about Dungeon mobs marked boss.

What are the threshold values for this effect's activation?

**Comment by twowTurtReynolds** [comment]
It looks to me (I don't have codebase access) that it is as simple as 
If boss == skull and playerclass == rogue then rogue avoidance = 0.

**Comment by RUTHINAS** [comment]
I haven't tanked raid bosses since after the change was implemented (the raid group took the hint and shelved me as too risky a tank), but if I'm reading this right (and if your assumption is correct), I should be fine to continue tanking raid bosses as long as the group size is "typical" (say, 20-40 for Molten Core, 12-20 for Zul'Gurub etc.)?

**Comment by twowTurtReynolds** [comment]
That, is up to you and the community to learn and figure out.

**Comment by RUTHINAS** [comment]
Seems to apply to Lower Halls of Karazhan bosses as well, in a group of size 10.

To be honest making avoidance plummet to zero in boss fights is unnecessarily much. Certainly there are other ways to make rogues not overpowered as avoidance tanks that are not making them completely unviable.

**Comment by balakethelock** [comment]
> It looks to me (I don't have codebase access) that it is as simple as If boss == skull and bosstarget == player and playerclass == rogue and people in instance < 6 then player avoidance = 0. Our core dev team can look to be sure and make adjustmeents to make it more fair to group dungeoneers like yourself who are playing as intended.

This may be personal opinion but they already gutted flourish, it's no longer possible to permanently tank these mobs anyway. The hackfix to avoidance that only affects rogues with skull targets is now obsolete and only does harm and no good so should be removed

**Comment by RUTHINAS** [comment]
Indeed. It also feels unnecessary to have an arbitrary division of "rogues can tank dungeons, but not raids". I can't really think of any other similar feature in the game (especially one that has no tooltip anywhere), even though it would make sense to have such features in place to differentiate PvE and PvP (in the style of crowd control lasting a limited time on players).

And it will just cause more issues down the line, when every new boss has to be flagged or unflagged for rogue avoidance removal depending on content.

**Comment by Sringaram** [comment]
The reason behind the original change which removed rogue's avoidance if you tried to "solo" raid bosses (it was never possible to begin with) was based on nebulous fabrications to begin with.
It's time to stop this hatred of the most neglected class on TWoW.
